src/wtc-autoplay-video.js

```diff
@@ -2,61 +2,213 @@
+   * 
+   * @constructor
+   * @param {DOMElement} element 
+   * @param {Object} options 
+   * @memberOf AutoplayVideo
+   */
   constructor(element, options) {
     super(element);
 
-    const defaults = { fullWidth: false, vpOn: 0 };
-    options = Object.assign({}, defaults, options);
+    // Bind anything that will be used a sa listener
+    this.onLoopCheck = this.onLoopCheck.bind(this);
+    this.init = this.init.bind(this);
+    this.onEnded = this.onEnded.bind(this);
+    this.onFrozen = this.onFrozen.bind(this);
+    this.animationCallback = this.viewportAnimationCallback.bind(this);
 
-    this.hasStarted = false;
-    this.initiated = false;
+    // Assign the provided options and coerce and HTML element properties as well.
+    options = Object.assign({}, defaults, options);
     this.options = {
       fullWidth: this.element.classList.contains('autoplay-video--fullscreen') || options.fullWidth,
-      vpOn: this.element.hasAttribute('data-vp-on') ? parseInt(this.element.getAttribute('data-vp-on')) : options.vpOn
+      vpOn: this.element.hasAttribute('data-vp-on') ? parseInt(this.element.getAttribute('data-vp-on')) : options.vpOn,
+      startAt: this.element.hasAttribute('data-autoplay-video--start-at') ? parseFloat(this.element.getAttribute('data-autoplay-video--start-at')) : options.startAt,
+      loopFrom: this.element.hasAttribute('data-autoplay-video--loop-from') ? parseFloat(this.element.getAttribute('data-autoplay-video--loop-from')) : options.loopFrom,
+      loopTo: this.element.hasAttribute('data-autoplay-video--loop-to') ? parseFloat(this.element.getAttribute('data-autoplay-video--loop-to')) : options.loopTo
     }
 
+    // If the video is set to full width, applu the class to the element.
     if (this.options.fullWidth && !this.element.classList.contains('autoplay-video--fullscreen')) {
       this.element.classList.add('autoplay-video--fullscreen');
     }
 
-    this._video = this.element.querySelector('.autoplay-video__video');
-    this._fallback = this.element.querySelector('.autoplay-video__fallback');
+    // Find the video and the fallback
+    this.video = this.element.querySelector('.autoplay-video__video');
+    this.fallback = this.element.querySelector('.autoplay-video__fallback');
 
-    this._video.muted = true;
-    this._video.setAttribute('playsinline', '');
-    this._video.setAttribute('muted', '');
+    // Set the required video properties for inline playing (thanks Chrome)
+    this.video.muted = true;
+    this.video.setAttribute('playsinline', '');
+    this.video.setAttribute('muted', '');
+    
+    // If we have a startAt property, update the video's time to that.
+    if(!isNaN(this.startAt) && this.startAt != null ) {
+      this.video.currentTime = this.startAt;
+    }
 
+    // Assign the class initialisation when the video can play
+    // @TODO clean this up. we have an else if with an init followed by an else with an init
```

Comment:
The final `else` used to have a check for the `canplay` event, but we ran into issues with Safari simply not firing it. Let's circle back to this and do some testing in Safari.